ambry-replication/src/main/java/com/github/ambry/replication/RemoteReplicaInfo.java

```diff
@@ -198,6 +204,23 @@ public boolean equals(Object obj) {
     return false;
   }
 
+  /**
+   * Get the meta data response information received for this replica in the most recent replication cycle.
+   * @return exchangeMetadataResponse contains the meta data response (missing keys, token info, local lag from remote, etc.).
+   */
+  synchronized ReplicaThread.ExchangeMetadataResponse getExchangeMetadataResponse() {
+    return exchangeMetadataResponse;
```

Comment:
When you working on future PR, keep in mind that `exchangeMetadataResponse` will be updated by multiple threads.